client/app/components/EventButtons.tsx

```diff
@@ -13,14 +13,14 @@ const EventButtons = ({
 }: {
   eventId: string;
   events: IEvent[];
-  forPage: 'results' | 'rankings' | 'data-entry';
+  forPage: 'results' | 'rankings' | 'competitions' | 'data-entry';
 }) => {
   const router = useRouter();
   const { id, singleOrAvg } = useParams();
   const searchParams = useSearchParams();
 
   const [selectedCat, setSelectedCat] = useState(
-    eventCategories.find((el) => events.find((e) => e.eventId === eventId).groups.includes(el.group)),
+    eventCategories.find((el) => events.find((e) => e.eventId === eventId)?.groups.includes(el.group)),
```

Comment:
Is it possible for an event to be undefined in the inner find here? Also, we should include the event categories when forPage = 'competitions'. Can you replace forPage === 'rankings' on line 52 with ['rankings', 'competitions'].includes(forPage)?

Reply:
Yes, it's possible for an event to be undefined, because on `competions` page we don't have any event/category selected by default.

Reply:
In that case it would be best to add `?? eventCategories[0]` at the very end of this line (before the ,), so that the first category is selected if the event id is empty